shuriken/mjolnir/include/mjolnir/MjolnIROps.td

```diff
@@ -0,0 +1,552 @@
+        returns a bool value for ControlFlowOps Dialect (I1 value)
+    }];
+
+    let arguments = (ins
+        DVM_BasicType:$lhs,
+        DVM_BasicType:$rhs
+    );
+
+    let results = (outs
+        I1:$res
+    );
+}
+
+def MjolnIR_Nop : MjolnIR_Op<"nop", [Pure]>
+{
+    let assemblyFormat = "attr-dict";
+}
+
+//===----------------------------------------------------------------------===//
+// Unary Operations
+//===----------------------------------------------------------------------===//
+def MjolnIR_MoveOp : MjolnIR_UnaryOpBase<"move">;
+def MjolnIR_CastOp : MjolnIR_UnaryOpBase<"cast">;
+
+def MjolnIR_Neg : MjolnIR_UnaryOpBase<"neg", [AllTypesMatch<["src", "dst"]>]>;
+def MjolnIR_Not : MjolnIR_UnaryOpBase<"not", [AllTypesMatch<["src", "dst"]>]>;
+
+//===----------------------------------------------------------------------===//
+// Load values operations
+//===----------------------------------------------------------------------===//
+def MjolnIR_LoadValue : MjolnIR_Op<"load-value", [Pure, ConstantLike]>
+{
+    let summary = "Load a constant value";
+
+    let description = [{
+        In order to generate a SSA value from a constant one we have to create
+        an instruction that has a value as attribute and returns a Value to any
+        of the types from Dalvik, in this way the generated value is already in
+        SSA value.
+
+        Examples:
+
+        ```mlir
+        // Example of data loaded
+        %0 = MjolnIR.load-value(33 : dvmint) : dvmint
+        ```
+    }];
+
+    let arguments = (ins
+        SI64Attr:$value
+    );
+
+    let results = (outs
+        DVM_BasicType:$res
+    );
+
+    let assemblyFormat = "`(` $value `)` attr-dict `:` type($res)";
+}
+
+//===----------------------------------------------------------------------===//
+// Load string operations
+//===----------------------------------------------------------------------===//
+def MjolnIR_LoadString : MjolnIR_Op<"load-string", [Pure, ConstantLike]>
+{
+    let summary = "Load a String into a String object";
+
+    let description = [{
+        In order to generate a SSA value from a loaded String, we have to create
+        an instruction which attribute will be the loaded string, and the return
+        will be a DVMObject with the attribute 'java/lang/String'.
+
+        Examples:
+
+        ```mlir
+        %0 = MjolnIR.load-string("ExampleString" : string) : dvmobject
+        ```
+    }];
+
+    let arguments = (ins
+        FlatSymbolRefAttr:$string,
+        UI32Attr:$stringRef
+    );
+
+    let results = (outs
+        DVM_BasicType:$res
+    );
+
+    let assemblyFormat = "`(` $string `)` attr-dict `:` type($res)";
+}
+
+//===----------------------------------------------------------------------===//
+// Binary operations
+//===----------------------------------------------------------------------===//
+def MjolnIR_AddOp : MjolnIR_BinaryOpBase<"add", [Commutative]>
```

Comment:
We need to check if these operations can be removed since we are using other dialects too!

Reply:
I can substitute for arith dialect

Reply:
maybe it is not needed, I just wanna know if the operations are being used or not, if not, we can remove it and make the file cleaner :)

Reply:
confirmed that it is needed